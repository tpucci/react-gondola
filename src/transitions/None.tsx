import React, { Children } from 'react';
import { View, StyleSheet } from 'react-native';

import { TransitionComponent } from './Transition';

export class None extends TransitionComponent {
  render() {
    if (!this.props.directionForward) {
      return null;
    }
    return (
      <View style={StyleSheet.absoluteFill}>
        {Children.only(this.props.children)}
      </View>
    );
  }
}
