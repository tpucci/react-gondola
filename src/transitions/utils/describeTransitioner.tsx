/* eslint no-undef:0 */

import React from 'react';
import TestRenderer from 'react-test-renderer';
import { Text } from 'react-native';

// @ts-ignore
export function describeTransitioner(Transitioner) {
  // @ts-ignore

  describe(Transitioner.displayName, () => {
    // @ts-ignore

    it('shows its child if direction is forward', () => {
      const testRenderer = TestRenderer.create(
        <Transitioner directionForward>
          <Text testID="child">I should be rendered</Text>
        </Transitioner>
      );
      // @ts-ignore

      expect(testRenderer.toJSON()).toMatchSnapshot();
    });
    // @ts-ignore

    it('hides its child if direction is backward', () => {
      const testRenderer = TestRenderer.create(
        <Transitioner directionForward={false}>
          <Text testID="child">I should not be rendered</Text>
        </Transitioner>
      );
      // @ts-ignore

      expect(testRenderer.toJSON()).toMatchSnapshot();
    });
    // @ts-ignore

    it('snapshots component when direction changed to backward', () => {
      const testRenderer = TestRenderer.create(
        <Transitioner directionForward>
          <Text testID="child">I should not be rendered</Text>
        </Transitioner>
      );
      testRenderer.update(
        <Transitioner directionForward={false}>
          <Text testID="child">I should not be rendered</Text>
        </Transitioner>
      );
      // @ts-ignore

      expect(testRenderer.toJSON()).toMatchSnapshot();
    });
    // @ts-ignore

    it('shows its child if direction changed to forward', () => {
      const testRenderer = TestRenderer.create(
        <Transitioner directionForward={false}>
          <Text testID="child">I should rendered</Text>
        </Transitioner>
      );
      testRenderer.update(
        <Transitioner directionForward>
          <Text testID="child">I should be rendered</Text>
        </Transitioner>
      );
      // @ts-ignore

      expect(testRenderer.toJSON()).toMatchSnapshot();
    });
  });
}
