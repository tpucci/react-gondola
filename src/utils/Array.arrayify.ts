export function arrayify<T>(arg: T | T[]): T[] {
  // @ts-ignore
  if (!arg.length) {
    // @ts-ignore
    return [arg];
  }
  // @ts-ignore
  return arg;
}
