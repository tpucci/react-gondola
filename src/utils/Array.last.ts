export function last<T>(arr: T[]): T | null {
  if (arr.length < 1) { return null; }
  return arr[arr.length - 1];
}
