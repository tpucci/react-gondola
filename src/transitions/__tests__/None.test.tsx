import { describeTransitioner } from '../utils/describeTransitioner';
import { None } from '../None';

describeTransitioner(None);
