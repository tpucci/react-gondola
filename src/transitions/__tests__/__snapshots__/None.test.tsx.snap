// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hides its child if direction is backward 1`] = `null`;

exports[`shows its child if direction changed to forward 1`] = `
<View
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <Text
    testID="child"
  >
    I should be rendered
  </Text>
</View>
`;

exports[`shows its child if direction is forward 1`] = `
<View
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <Text
    testID="child"
  >
    I should be rendered
  </Text>
</View>
`;

exports[`snapshots component when direction changed to backward 1`] = `null`;
